testutil/assert.go

```diff
@@ -0,0 +1,131 @@
+package testutil
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+type miniTB interface {
+	Helper()
+	Fatalf(string, ...any)
+}
+
+// formatSuffix builds an optional suffix from msgAndArgs
+func formatSuffix(msgAndArgs ...any) string {
+	switch len(msgAndArgs) {
+	case 0:
+		return ""
+	case 1:
+		return " - " + fmt.Sprintf("%+v", msgAndArgs[0])
+	default:
+		return " - " + fmt.Sprintf(msgAndArgs[0].(string), msgAndArgs[1:]...)
+	}
+}
+
+// Comparable types (numbers, strings, pointers to the same object, etc.).
+func AssertEqual[T comparable](tb miniTB, got, want T, msgAndArgs ...any) {
+	tb.Helper()
+	if got != want {
+		suffix := formatSuffix(msgAndArgs...)
+		tb.Fatalf("got %v; want %v%s", got, want, suffix)
+	}
+}
```

Comment:
Can't this work with any?


```suggestion
// Comparable types (numbers, strings, pointers to the same object, etc.).
func AssertEqual(tb miniTB, got, want any, msgAndArgs ...any) {
	tb.Helper()
	if got != want {
		suffix := formatSuffix(msgAndArgs...)
		tb.Fatalf("got %v; want %v%s", got, want, suffix)
	}
}
```

Reply:
If not, then adding as many asserter needed would be better than bumping Go minimal version

Here I'm suggesting adding

AssertEqualInt
AssertEqualFloat64
...

Reply:
Fixed